File: api/Controllers/ClienteController.cs
using api.DAO;
using api.Models;
using api.Repositories;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using System.Text.Json.Nodes;

namespace api.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class ClienteController : ControllerBase
    {
        private readonly ClienteDAO _clienteDAO;

        public ClienteController()
        {
            _clienteDAO = new ClienteDAO();
        }

        [Route("cadastro")]
        [HttpPost]
        public void Cadastrar([FromBody] Cliente cliente)
        {
            _clienteDAO.Cadastrar(cliente);
        }

        [Route("login")]
        [HttpPost]
        public ActionResult<Cliente> Login([FromBody] LoginDados _params)
        {
            var cliente = _clienteDAO.Login(_params);

            if (cliente == null) return BadRequest("email ou senha incorreta");

            return Ok(cliente);
        }

        [Route("buscarPorConta")]
        [HttpGet]
        public ActionResult<Cliente> BuscaPorConta([FromQuery] long conta)
        {
            var cliente = _clienteDAO.BuscaPorConta(conta);

            if (cliente == null) return BadRequest("conta inválida");

            return Ok(cliente);
        }
    }
}

File: api_node/src/controllers/cliente-controller.ts
import { Request, Response } from 'express';
import { IClinteRepository } from '../repositories/cliente-repository';
import { ClienteDAO } from '../dao/cliente-dao';

export class ClienteController {
  private readonly _clienteRepository: IClinteRepository;

  constructor(){
    this._clienteRepository = new ClienteDAO();
  }

  public async login(req: Request, res: Response) {
    try {
      const data = req.body;
      const resposta = await this._clienteRepository.login(data);

      if(resposta)
        return res.status(200).json(resposta);

      return res.status(400).json({ mensagem: 'conta ou senha inválida' })
    }
    catch(error) {
      res.status(400).json({ error });
    }
  }

  public async buscarPorId(req: Request, res: Response) {
    try {
      const idCliente = req.query.idCliente as string;
      const resposta = await this._clienteRepository.buscarPorId(parseInt(idCliente));
      res.status(200).json(resposta);
    }
    catch(error) {
      res.status(400).json({ error });
    }
  }

  public async buscarPorConta(req: Request, res: Response) {
    try {
      const conta = req.query.conta as string;
      const resposta = await this._clienteRepository.buscarPorConta(parseInt(conta));
      
      if(resposta)
        return res.status(200).json(resposta);

      return res.status(400).json({ mensagem: 'conta inválida' })
    }
    catch(error) {
      res.status(400).json({ error });
    }
  }
}